src/components/Question.js

```diff
@@ -26,14 +26,20 @@ class Question extends Component {
       answerSubmitted: null,
       error: false,
       correctAnswer: null,
-      explanationRequested: false
+      explanationRequested: false,
     };
   }
 
   calcNewScore(isCorrect) {
     const { updateScore, score, bookId, chapterId, index } = this.props;
     const questionIndex = index - 1;
-    const newScore = score.map((book, index) => {
+
+    /* check if current question was already answered correctly */
+    const alreadyAnswered =
+      score.books[bookId].chapters[chapterId].questions[questionIndex].answered;
```

Comment:
Tests throw `TypeError` when trying to access `score.books[bookId].chapters`. No problems when running the app. Unsure how to proceed?

Reply:
If you look in the __tests__/Question.js file, you will notice that the test has the following props:

```javascript
const generateTestProps = question => {
  return {
    question,
    baseUrl,
    index,
    numberOfQuestions,
    score,
    updateScore,
  };
};
```

This means that it doesn't have `bookId` and `chapterId` as props when it renders `Question.js` for the tests.

Also, the sidebar looks awesome!

Reply:
Wow, that makes perfect sense. You just put 2 and 2 together for me. I was staring at the file for a long time looking for `chapterId` somewhere in the failing tests... and now I'm interested in testing. Gonna have to look at Enzyme over the weekend. 

Anyway, is it correct for me to assume that I just need to provide those props to `generateTestProps`, since it looks like all the tests only need to cover a few books/chapters? In general, don't hard-code stuff like that in, but for these tests as they are now, it seems fine?

Reply:
Yup! If you take a quick look at the test file...

https://github.com/austintackaberry/ydkjs-exercises/blob/master/src/__tests__/Question.js

...I think it will all come together. Basically, since we are doing a "shallow" render of `Question.js` for our tests, we need to send it hardcoded props, so that it can function normally. If we don't generate all the props that are needed, then it will break.

Reply:
Makes sense to me, also actually really interesting. I feel like I've been missing out by not incorporating testing into my personal projects. Maybe in like two or three PRs I'll be writing some tests for this project. Thanks for the help!